extensions/dacpac/src/test/dacFxSummaryPage.test.ts

```diff
@@ -0,0 +1,83 @@
+	beforeEach(async function (): Promise<void> {
+		wizard = new DataTierApplicationWizard();
+		wizard.model = <DacFxDataModel>{};
+		wizard.model.server = undefined;
+		wizard.setPages();
+		wizard.configureButtons();
+		testContext = createContext();
+	});
+
+	it('DacFx Summary Page should start correctly for deploying to an existing database', async () => {
+		wizard.selectedOperation = Operation.deploy;
+		wizard.model.upgradeExisting = true;
+
+		const summaryPage = await validatePageCreation();
+		should(summaryPage.WizardState.wizard.generateScriptButton.hidden).equal(false, 'Generate script button should not be hidden when the operation is deploy and upgrade existing is true');
+		await validateOnPageLeave(summaryPage);
+	});
+
+	it('DacFx Summary Page should start correctly for deploying to a new database', async () => {
+		wizard.selectedOperation = Operation.deploy;
+		wizard.model.upgradeExisting = false;
+
+		const summaryPage = await validatePageCreation();
+		should(summaryPage.WizardState.wizard.generateScriptButton.hidden).equal(true, 'Generate script button should be hidden when deploying to a new database');
+		await validateOnPageLeave(summaryPage);
+	});
+
+	it('DacFx Summary Page should start correctly for extract', async () => {
+		wizard.selectedOperation = Operation.extract;
+
+		const summaryPage = await validatePageCreation();
+		should(summaryPage.WizardState.wizard.generateScriptButton.hidden).equal(true, 'Generate script button should be hidden when extracting');
+		await validateOnPageLeave(summaryPage);
+	});
+
+	it('DacFx Summary Page should start correctly for import', async () => {
+		wizard.selectedOperation = Operation.import;
+
+		const summaryPage = await validatePageCreation();
+		should(summaryPage.WizardState.wizard.generateScriptButton.hidden).equal(true, 'Generate script button should be hidden when importing');
+		await validateOnPageLeave(summaryPage);
+	});
+
+	it('DacFx Summary Page should start correctly for exporting', async () => {
+		wizard.selectedOperation = Operation.export;
+
+		const summaryPage = await validatePageCreation();
+		should(summaryPage.WizardState.wizard.generateScriptButton.hidden).equal(true, 'Generate script button should be hidden when exporting');
+		await validateOnPageLeave(summaryPage);
+	});
+
+	async function validatePageCreation(): Promise<TestDacFxSummaryPage> {
+		const summaryPage: TestDacFxSummaryPage = new TestDacFxSummaryPage(wizard, wizard.pages.get(PageName.selectOperation).wizardPage, wizard.model, testContext.viewContext.view);
+		const onPageStart = await summaryPage.start();
+		const onPageEnter = await summaryPage.onPageEnter();
+		should(onPageStart).equal(true);
+		should(onPageEnter).equal(true);
+
+		return summaryPage;
+	}
+
+	async function validateOnPageLeave(summaryPage: TestDacFxSummaryPage): Promise<void> {
+		const onPageLeave = await summaryPage.onPageLeave();
+		should(onPageLeave).equal(true);
```

Comment:
Is there anything else that we can verify (probably for each page), to make sure that some things (may be some textbox/label) exist on the page after its created/left?

Reply:
added a couple checks to make sure the table data is populated